server/service/src/item/item.rs

```diff
@@ -19,12 +21,89 @@ pub fn get_items(
     let connection = connection_manager.connection()?;
     let repository = ItemRepository::new(&connection);
 
+    let filter = filter
```

Comment:
All this logic looks great! It's pretty involved - I think definitely worthy of some tests, both to help future devs understand the expected behaviour, and ensure any changes in future don't break functionality.

Happy to pair on what some of these cases could be/setting up the tests if you would like!